### PR TITLE
Early July Decomp

### PR DESCRIPTION
**Pull request by SethBarberee**
* More documenting of music
* Kecleon decomp
* Kangaskhan decomp
* Wonder mail decomp
* Friend list menu decomp

### CODE REVIEW

File: include/dungeon_entity.h [hunk 1]
```diff
@@ -162,7 +162,8 @@ struct DungeonEntityData
     /* 0xF5 */ bool8 exposedStatus;
     /* 0xF6 */ bool8 isColorChanged;
     /* 0xF7 */ bool8 isBoss;
-    u8 fillF8[0xFA - 0xF8];
+    /* 0xF8 */ u8 unkF8;
+    /* 0xF8 */ u8 unkF9;
```
Comment (AnonymousRandomPerson) [inline include/dungeon_entity.h:166]: Update this comment to 0xF9.

File: src/music.c [hunk 2]
```diff
@@ -1,44 +1,1127 @@
...
+            EnableInterrupts();
+        }
+    }
+}
+
+bool8 IsFanfareSEPlaying(u16 songIndex)
+{
+  u32 playerIndex;
+  struct PMDMusicPlayer *musicPlayer;
+  register u32 songIndex_u32 asm("r4");
```
Comment (AnonymousRandomPerson) [inline src/music.c:906]: Missing #NONMATCHING?

**Review by AnonymousRandomPerson — CHANGES_REQUESTED** [review]

File: src/music.c (hunk 2 above)
Comment (SethBarberee) [reply src/music.c:906]: This does match however, the main thing is the register pinning. Most of pokeemerald/pokefirered follow the same. My current philosophy is to use NONMATCHING when `asm_unified` is used.

File: include/dungeon_entity.h (hunk 1 above)
Comment (SethBarberee) [reply include/dungeon_entity.h]: Fixed

**Review by AnonymousRandomPerson — APPROVED** [review]

File: src/music.c (hunk 2 above)
Comment (SethBarberee) [reply src/music.c:906]: So I was wrong on the above (from GriffinR)
`fwiw, when pokeemerald/pokefirered still had uses of the register keyword they were wrapped in NONMATCHING`. Will update this function with `NONMATCHING` just to follow convention.

**Review by AnonymousRandomPerson — APPROVED** [review]

**Pull request merged**